File: Unit_tests_lua/EX-01-compare_directories/Claude.lua
function compare_directories(dir1, dir2)

    local report = {
      Passed = true,
      Failed_Count = 0,
      Failed_Tests = {}
    }
  
    -- Loop through all files in dir1
    for _, file1 in ipairs(list_files(dir1)) do
  
      -- Get corresponding file in dir2
      local file2 = get_file(dir2, get_filename(file1))
  
      -- Compare files
      if file1.modified < file2.modified then
        -- File in dir1 is older
        report.Failed_Location = "dir1" 
      elseif file1.modified > file2.modified then
        -- File in dir2 is older
        report.Failed_Location = "dir2"
      end
  
      -- Check if files are identical
      if file1.content ~= file2.content then
        
        -- Update report
        report.Passed = false
        report.Failed_Count = report.Failed_Count + 1
  
        table.insert(report.Failed_Tests, {
          Failed_Files = {file1.name, file2.name},
          Failure_Location = report.Failed_Location
        })
  
      end
  
    end
  
    return report
  
  end

File: Unit_tests_python/EX-01-compare_directories/Claude.py
import os
from datetime import datetime

def compare_directories(dir1, dir2):
    
    report = {
        'Passed': True,
        'Failed_Count': 0,
        'Failed_Tests': []
    }
    
    for root, dirs, files in os.walk(dir1):
        for file in files:
            file1 = os.path.join(root, file)
            file2 = os.path.join(dir2, file)
            
            if not os.path.exists(file2):
                report['Passed'] = False
                report['Failed_Count'] += 1
                report['Failed_Tests'].append({
                    'Failed_Files': [file],
                    'Failure_Location': 'dir2' 
                })
                
            else:
                stat1 = os.stat(file1)
                stat2 = os.stat(file2)
                
                if stat1.st_mtime < stat2.st_mtime:
                    failure_location = 'dir1'
                else:
                    failure_location = 'dir2'
                    
                if stat1.st_size != stat2.st_size:
                    report['Passed'] = False
                    report['Failed_Count'] += 1
                    report['Failed_Tests'].append({
                        'Failed_Files': [file],
                        'Failure_Location': failure_location
                    })
                
    return report